src/main.cpp

```diff
@@ -981,6 +981,36 @@ bool ContextualCheckTransaction(
             // https://zips.z.cash/zip-0213#specification
             uint256 ovk;
             for (const OutputDescription &output : tx.vShieldedOutput) {
+                bool zip_212_enabled;
+                libzcash::SaplingPaymentAddress zaddr;
+                CAmount value;
+
+            // EoS height for 5.3.3 and 5.4.2 is 2121024.
+            if (nHeight >= 2121200) {
+                try {
+                    auto decrypted = wallet::try_sapling_output_recovery(
+                        *chainparams.RustNetwork(),
+                        nHeight,
+                        ovk.GetRawBytes(),
+                        {
+                            output.cv.GetRawBytes(),
+                            output.cmu.GetRawBytes(),
+                            output.ephemeralKey.GetRawBytes(),
+                            output.encCiphertext,
+                            output.outCiphertext,
+                        });
+                    zip_212_enabled = decrypted->zip_212_enabled();
```

Comment:
```suggestion
                    zip_212_enabled = decrypted->zip_212_enabled();
                    // Guaranteed by try_sapling_output_recovery (which indirectly calls
                    // parse_note_plaintext_without_memo_ovk) for this block height.
                    assert(zip_212_enabled);
```
```

Reply:
I agree that this holds for the code as of this PR, but we'd need to remove the assertion when the old code is removed, so I don't see a strong motivation for adding it now (it would only be a temporary consistency check).